jupyterlab_gitsync/src/service/notebook_file.ts

```diff
@@ -1,31 +1,40 @@
 import { DocumentRegistry } from '@jupyterlab/docregistry';
 import { NotebookPanel, Notebook, NotebookModel } from '@jupyterlab/notebook';
 import { ISignal, Signal } from '@lumino/signaling';
+import { ContentsManager, Contents } from '@jupyterlab/services'
 
-import { ContentsManager } from '@jupyterlab/services';
-
-import { IFile, IResolver } from './tracker';
+import { IFile } from './tracker';
 import { NotebookResolver } from './notebook_resolver';
 
 const fs = new ContentsManager();
 
-// TO DO (ashleyswang): implement most functionality for NotebookFile
-// mostly a placeholder file with outline of needed functions
-// so compiler doesn't complain
-
 export class NotebookFile implements IFile {
   widget: NotebookPanel;
+  content: Notebook;
   context: DocumentRegistry.Context;
-  resolver: IResolver;
+  resolver: NotebookResolver;
+  view: {
+    left: number,
+    top: number
+  }
+  cursor: {
+    index: number,
```

Comment:
my curiosity, what is a cursor index?

Reply:
`index` is the cell index in which the cursor lies. Since each cell has it's own editor, we need to get the index of the cell as well as it's line and character (column) position.